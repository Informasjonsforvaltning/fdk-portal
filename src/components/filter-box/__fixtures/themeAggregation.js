export const themeAggregation = {
  buckets: [
    { key: 'GOVE', count: 104 },
    { key: 'EDUC', count: 11 },
    { key: 'HEAL', count: 8 },
    { key: 'Ukjent', count: 7 },
    { key: 'AGRI', count: 1 },
    { key: 'TEST', count: 1 }
  ]
};
