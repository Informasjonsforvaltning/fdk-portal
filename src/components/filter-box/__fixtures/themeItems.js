export const themeItems = {
  AGRI: {
    id: 'http://publications.europa.eu/resource/authority/data-theme/AGRI',
    code: 'AGRI',
    startUse: '2015-10-01',
    title: {
      pt: 'Agricultura, pesca, silvicultura e alimentação',
      bg: 'Селско стопанство, рибарство, горско стопанство и храни'
    },
    conceptSchema: {
      id: 'http://publications.europa.eu/resource/authority/data-theme',
      title: { en: 'Dataset types Named Authority List' }
    }
  }
};
