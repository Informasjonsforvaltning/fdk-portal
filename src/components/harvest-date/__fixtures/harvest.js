export const harvest = {
  firstHarvested: '2018-06-25T19":51:05+0200',
  lastHarvested: '2018-07-09T14:40:37+0200',
  lastChanged: '2018-07-09T14:40:37+0200',
  changed: [
    '2018-06-25T19:51:05+0200',
    '2018-06-25T19:51:12+0200',
    '2018-06-26T09:16:46+0200',
    '2018-06-28T09:22:41+0200',
    '2018-06-28T09:52:24+0200',
    '2018-06-29T15:25:53+0200',
    '2018-07-03T10:59:38+0200',
    '2018-07-09T14:40:37+0200'
  ]
};
