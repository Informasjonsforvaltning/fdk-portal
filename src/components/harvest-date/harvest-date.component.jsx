import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Moment from 'react-moment';
import localization from '../../lib/localization';
import './harvest-date.scss';

const renderHarvested = harvest => {
  if (!_.get(harvest, 'firstHarvested')) {
    return null;
  }
  return (
    <span className="d-flex flex-wrap align-self-center">
      {localization.dataset.firstHarvested}&nbsp;
      <Moment format="DD.MM.YYYY">{harvest.firstHarvested}</Moment>&nbsp;/&nbsp;
    </span>
  );
};

const renderLastChanged = harvest => {
  if (!_.get(harvest, 'lastChanged')) {
    return null;
  }
  return (
    <span className="d-flex flex-wrap align-self-center">
      {localization.dataset.lastChanged}&nbsp;
      <Moment format="DD.MM.YYYY">{_.get(harvest, 'lastChanged')}</Moment>
    </span>
  );
};

export const HarvestDate = props => {
  const { harvest } = props;
  if (!harvest) {
    return null;
  }

  return (
    <div className="d-flex flex-wrap">
      {renderHarvested(harvest)}
      {renderLastChanged(harvest)}
    </div>
  );
};

HarvestDate.defaultProps = {
  harvest: null
};

HarvestDate.propTypes = {
  harvest: PropTypes.object
};
