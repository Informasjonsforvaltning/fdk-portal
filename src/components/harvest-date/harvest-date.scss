@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-detail-date {
    color: black !important;
}
