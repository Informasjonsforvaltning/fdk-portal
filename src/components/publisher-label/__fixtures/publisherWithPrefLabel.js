export const publisherWithPrefLabel = {
  type: 'no.dcat.datastore.domain.dcat.Publisher',
  valid: false,
  uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
  id: '910244132',
  name: 'RAMSUND OG ROGNAN REVISJON',
  orgPath: '/ANNET/910244132',
  prefLabel: {
    no: 'Ramsund og Rognan revisjon'
  }
};
