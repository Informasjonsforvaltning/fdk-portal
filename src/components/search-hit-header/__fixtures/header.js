export const header = {
  title: 'Test tittel',
  publisherLabel: 'Eies av',
  publisher: {
    id: '974760673'
  }
};
