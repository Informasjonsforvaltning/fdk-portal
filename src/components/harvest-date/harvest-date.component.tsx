import React, { FC } from 'react';
import Moment from 'react-moment';

import localization from '../../lib/localization';

import './harvest-date.scss';

interface Props {
  harvest?: Record<string, any>;
}

export const HarvestDate: FC<Props> = ({ harvest }) =>
  harvest ? (
    <div className='d-flex flex-wrap'>
      {harvest?.firstHarvested && (
        <span className='d-flex flex-wrap align-self-center'>
          {localization.dataset.firstHarvested}&nbsp;
          <Moment format='DD.MM.YYYY'>{harvest.firstHarvested}</Moment>
          &nbsp;/&nbsp;
        </span>
      )}
      {harvest?.lastChanged && (
        <span className='d-flex flex-wrap align-self-center'>
          {localization.dataset.lastChanged}&nbsp;
          <Moment format='DD.MM.YYYY'>{harvest.lastChanged}</Moment>
        </span>
      )}
    </div>
  ) : null;
