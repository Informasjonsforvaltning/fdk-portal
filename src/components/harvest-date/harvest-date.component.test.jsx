import React from 'react';
import { shallow } from 'enzyme';
import { HarvestDate } from './harvest-date.component';
import { harvest } from './__fixtures/harvest';

test('should render HarvestDate with no props', () => {
  const wrapper = shallow(<HarvestDate />);
  expect(wrapper).toMatchSnapshot();
});

test('should render HarvestDate with harvest dates', () => {
  const wrapper = shallow(<HarvestDate harvest={harvest} />);
  expect(wrapper).toMatchSnapshot();
});
