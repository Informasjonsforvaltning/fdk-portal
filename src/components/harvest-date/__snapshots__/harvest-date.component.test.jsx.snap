// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HarvestDate with harvest dates 1`] = `
<div
  className="d-flex flex-wrap"
>
  <span
    className="d-flex flex-wrap align-self-center"
  >
    Published in The National Data Catalog
     
    <t
      ago={false}
      calendar={false}
      decimal={false}
      element="time"
      filter={[Function]}
      format="DD.MM.YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    >
      2018-06-25T19":51:05+0200
    </t>
     / 
  </span>
  <span
    className="d-flex flex-wrap align-self-center"
  >
    Last changed in The National Data Catalog
     
    <t
      ago={false}
      calendar={false}
      decimal={false}
      element="time"
      filter={[Function]}
      format="DD.MM.YYYY"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    >
      2018-07-09T14:40:37+0200
    </t>
  </span>
</div>
`;

exports[`should render HarvestDate with no props 1`] = `""`;
